### PR TITLE
[Bug] Some fresh start fixes

### PR DESCRIPTION
**Pull request by Tempo-anon**
## What are the changes?
* Starters in fresh start are not hardcoded as 3 cost
* Mini Black Hole disabled in fresh start
* No catching allowed in the last biome in fresh start

## Why am I doing these changes?
Makes fresh start more inline to a true fresh start. Note that there are still some edge cases involving dexAttr which are not easily solved.

## What did change?
* Created 2 new helper functions in game-mode.ts
* Added a fresh start check for mini black hole and eviolite weight
* Added a fresh start check for catching in the last biome

## How to test the changes?
* Change starter costs from 3 and see if it's correctly reflected in fresh start
* Change MBH weight super high and see if it will spawn in fresh start
* Try catching in the last biome in fresh start

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [ ] Have I considered writing automated tests for the issue?
- [ ] Have I tested the changes (manually)?
    - [ ] Are all unit tests still passing? (`npm run test`)
- [ ] Are the changes visual?
  - [ ] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Comment by Xavion3** [comment]
I'd really prefer if we just had new challenge types, because challenges wanting to modify whether you can catch a pokemon or whether you can get a particular item are both useful for other challenges and also it means we don't need to be special casing things.

**Comment by damocleas** [comment]
> I'd really prefer if we just had new challenge types, because challenges wanting to modify whether you can catch a pokemon or whether you can get a particular item are both useful for other challenges and also it means we don't need to be special casing things.

I have no idea what you mean by this at all.

**Comment by Xavion3** [comment]
It's technical implementation things, I'm just saying I'd prefer it to be coded a different way so it's in line with how challenges do things instead of a hacky short term fix we'll want to change later anyway.

It has no impact on whether or not I think these features are warranted.

File: src/data/challenge.ts [hunk 1]
```diff
@@ -636,7 +636,7 @@ export class FreshStartChallenge extends Challenge {
 
   applyStarterCost(species: Species, cost: Utils.NumberHolder): boolean {
     if (defaultStarterSpecies.includes(species) && cost.value !== 3) {
-      cost.value = 3;
+      cost.value = speciesStarters[species];
```
Comment (DayKev) [inline src/data/challenge.ts:639]: Is the `if` even needed anymore?

File: src/modifier/modifier-type.ts [hunk 2]
```diff
@@ -1623,7 +1623,7 @@ const modifierPool: ModifierPool = {
     new WeightedModifierType(modifierTypes.MULTI_LENS, 18),
     new WeightedModifierType(modifierTypes.VOUCHER_PREMIUM, (party: Pokemon[], rerollCount: integer) => !party[0].scene.gameMode.isDaily && !party[0].scene.gameMode.isEndless && !party[0].scene.gameMode.isSplicedOnly ? Math.max(5 - rerollCount * 2, 0) : 0, 5),
     new WeightedModifierType(modifierTypes.DNA_SPLICERS, (party: Pokemon[]) => !party[0].scene.gameMode.isSplicedOnly && party.filter(p => !p.fusionSpecies).length > 1 ? 24 : 0, 24),
-    new WeightedModifierType(modifierTypes.MINI_BLACK_HOLE, (party: Pokemon[]) => party[0].scene.gameData.unlocks[Unlockables.MINI_BLACK_HOLE] ? 1 : 0, 1),
+    new WeightedModifierType(modifierTypes.MINI_BLACK_HOLE, (party: Pokemon[]) => (!party[0].scene.gameMode.isFreshStartChallenge() && party[0].scene.gameData.unlocks[Unlockables.MINI_BLACK_HOLE]) ? 1 : 0, 1),
```
Comment (xsn34kzx) [inline src/modifier/modifier-type.ts:1626]:
```suggestion
    new WeightedModifierType(modifierTypes.MINI_BLACK_HOLE, (party: Pokemon[]) => {
      const { scene } = party[0];
      return scene.gameMode.isFreshStartChallenge() && scene.gameData.unlocks[Unlockables.MINI_BLACK_HOLE]) ? 1 : 0
    }, 1),
```
+1 to this comment (https://github.com/pagefaultgames/pokerogue/pull/3420#discussion_r1711932591), we can apply the same syntax to this entry to be a bit cleaner.

**Review by xsn34kzx — COMMENTED** [review]
Pretty much cosigning flx's syntax comments. Since this is more or less a band-aid for now, I'll probably look into making disabling unlocks, etc. apart of Fresh Start in general like Xavion was saying.

**Pull request merged**